### PR TITLE
Geometry_Engine: Tolerance introduced to the check of 2-dimensionality of the input point set

### LINKED ISSUE #3288: Geometry_Engine: Apply tolerance to preliminary check in FitLine
#### Description
~~https://github.com/BHoM/BHoM_Engine/pull/3280 changes the `FitLine` method in a way that makes `tolerance` input redundant. It would be good to remove it, rather in a separate PR to avoid the valuable feature being blocked due to versioning and cross dependencies.~~

Actually, the tolerance is needed to verify whether the input is at least 2-dimensional (in simple words, whether all input points do not overlap). Therefore removal of tolerance is not a solution, we rather need to make use of it instead.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #3288

### Test files

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FBHoM%5FEngine%2FGeometry%5FEngine%2F%233293%2DMakeUseOfToleranceInFitLine&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4)

### CODE REVIEW

**Comment by pawelbaran** [comment]
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check required

**Review by peterjamesnugent — COMMENTED** [review]
Happy with the test script, and that it works as expected. Very clear and easy to follow.

Would it be worth adding an error message if we are going to return null?

**Comment by pawelbaran** [comment]
Thanks @peterjamesnugent!

> Would it be worth adding an error message if we are going to return null?

Concerning the above question, I would rather not do that because e.g. `IsCollinear` is meant to return `true` in case of `FitLine() == null`, assuming that coincident points are collinear. Would that make sense to you?

File: Geometry_Engine/Compute/FitLine.cs [hunk 1]
```diff
@@ -37,13 +37,13 @@ public static partial class Compute
 
         [Description("Fits a line into a set of points using Orthogonal Least Squares algorithm.")]
```
Comment (peterjamesnugent) [inline Geometry_Engine/Compute/FitLine.cs:38]:
```suggestion
        [Description("Fits a line into a set of points using Orthogonal Least Squares algorithm. Will return null if the number of non-duplicate points is less than two.")]
```

Maybe something like this if we don't want to raise an error?

**Review by peterjamesnugent — APPROVED** [review]
Happy changes have been implementd and code is unchanged.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: unit-tests

**Pull request merged**